### PR TITLE
Campaign stuff

### PR DESCRIPTION
**Pull request by TheTh1rt33nth**
Pls fix easyjson(((

### CODE REVIEW

File: internal/app/CampaignManagement.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+func AddPlayedBattle (name string, campaignID int, player1 int, player2 int, victor int, lore string)
+{
+  battlereport=model.CampaignEvent{
+    Name: name,
+    CampaignID: campaignID,
+    Player1: player1,
+    Player2: player2,
+    Victor: victor,
+  }
+  if lore!=nil{
```
Comment (f0m41h4u7) [inline internal/app/CampaignManagement.go]: hmmm i thought string doesn't have nil value...

File: internal/app/rostermanagement.go [hunk 2]
```diff
@@ -52,6 +54,30 @@ func CountCP(roster *model.Army) int {
 	return cp
 }
 
+func RPonSL (armyID int){
```
Comment (f0m41h4u7) [inline internal/app/rostermanagement.go]: wow so understandable )

File: internal/app/CampaignManagement.go (hunk 1 above)
Comment (f0m41h4u7) [reply internal/app/CampaignManagement.go]: yes, it actually doesn't

File: internal/app/rostermanagement.go (hunk 2 above)
Comment (TheTh1rt33nth) [reply internal/app/rostermanagement.go]: Spending requisition points on supply limit))

Comment (f0m41h4u7) [reply internal/app/rostermanagement.go]: во имя Омниссии это же так obvious очевидно понятно ваще жесть !

**Review by f0m41h4u7 — CHANGES_REQUESTED** [review]
rename the function pleas !!
or at least write a comment above it

File: internal/app/roster_management.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	rosterObj := dummyRoster // but actually ask DB for real roster by armyID
+	if rosterObj.RequisitionPoints > 0 {
+		rosterObj.RequisitionPoints -= 1
+		rosterObj.SupplyLimit += 1
+	} else {
+		fmt.Println("Not enough RP. Fight harder, commander!")
+	}
+}
+
+func CPbyPL(pl int) int {
```
Comment (f0m41h4u7) [inline internal/app/roster_management.go]: eto chto

Comment (f0m41h4u7) [reply internal/app/roster_management.go]:
```suggestion
func PLtoCP(pl int) int {
```

**Review by f0m41h4u7 — APPROVED** [review]

File: internal/app/roster_management.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	pl := 0
+	dummydetaches := []model.Detachment{} //get actual detaches from DB
+	if len(dummydetaches) == 0 {
+		return 0
+	}
+	for _, d := range dummydetaches {
+		cp -= d.CpCost
+		pl += d.PowerLevel
+	}
+	cp += CPbyPL(pl)
```
Comment (f0m41h4u7) [inline internal/app/roster_management.go]:
```suggestion
	cp += PLtoCP(pl)
```

**Pull request merged**